### PR TITLE
feat(setup): Initial code for context api setup with incorporated review comments

### PR DESCRIPTION
**Pull request by vishalverma26**

### CODE REVIEW

File: templates/base/store/global/Context.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// Creating global context
+export const Context = React.createContext();
+
+let [state, dispatch] = useReducer(reducer, initialState);
+const GlobalContextProvider = props => {
+    /**
+     * Once the app has been Mounted / Loaded successfully
+     * We shall update the "Loading" property of state to false
+     */
+    useEffect(() => {
```
Comment (vinodloha) [inline templates/base/store/global/Context.jsx:19]: Take this useEffect

File: templates/base/store/global/Context.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ *  */
+
+import React, { useReducer, useEffect } from 'react';
+import reducer, { initialState } from './reducer';
+import { APP_LOADED } from './actions';
+
+// Creating global context
+export const Context = React.createContext();
+
+let [state, dispatch] = useReducer(reducer, initialState);
```
Comment (vinodloha) [inline templates/base/store/global/Context.jsx:13]: This is context definition file, here we will not execute actions and useReducer

I think you miss understood my comment on uses of these actions that was from reducer uses as it was empty

File: templates/base/store/global/actions.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export const APP_ERROR = 'APP_ERROR';
+export const APP_LOADED = 'APP_LOADED';
```
Comment (vinodloha) [inline templates/base/store/global/actions.js:2]: Delete this App_Loaded one

File: templates/base/store/global/constants.js [hunk 4]
```diff
@@ -0,0 +1 @@
+export const DUMMY_CONSTANT = 'DUMMY_CONSTANT';
```
Comment (vinodloha) [inline templates/base/store/global/constants.js:1]: Bring the consts here

File: templates/base/store/global/reducer.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+// @Initial State
+export const initialState = {
+    loading: true,
+    error: false
+}
+
+const globalReducer = (state = initialState, action) => {
+    switch(action.type) {
+        case APP_ERROR:
+            return {
```
Comment (vinodloha) [inline templates/base/store/global/reducer.js:12]: Use the payload from action.payload, no point hard coding an error state to always true

File: templates/base/store/global/reducer.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+                ...state,
+                error: true
+            }
+        case APP_LOADED:
```
Comment (vinodloha) [inline templates/base/store/global/reducer.js:16]: Delete this one

File: templates/base/store/global/reducer.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import { APP_ERROR } from './actions';
```
Comment (vinodloha) [inline templates/base/store/global/reducer.js:1]: This is wrong, you need to use the same constant that used in action 
UPDATE_APP_ERROR

File: templates/base/store/global/reducer.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+// @Initial State
+const initialState = {
+    errorInfo: {},
+    isError: false
+}
+
+const globalReducer = (state = initialState, action) => {
+    switch(action.type) {
+        case APP_ERROR:
```
Comment (vinodloha) [inline templates/base/store/global/reducer.js:11]: here too Same action UPDATE_APP_ERROR

File: templates/base/store/global/reducer.js (hunk 7 above)
Comment (vishalverma26) [reply templates/base/store/global/reducer.js]: fixed

Comment (vishalverma26) [reply templates/base/store/global/reducer.js]: fixed.

File: templates/base/store/global/reducer.js (hunk 8 above)
Comment (vishalverma26) [reply templates/base/store/global/reducer.js]: fixed

**Pull request merged**